### DISCUSSION TITLE
Any way to access the loss function within a optimizer?

### DISCUSSION THREAD

**Discussion by Ethan74884** [keras-team/keras#19823] [Q&A]
I am trying to implement an idea I had for a optimizer, but it requires calculating the loss during the parameter update, Google has nothing and looking and I failed to find any documentation about such a possibility, any way to do it?

**Comment by abhaskumarsinha** [comment]
What does that mean?
I didn't get it, calculating loss *during parameter update*?

Usually, the loss function is a function that measures the loss value through the output of the model and actual output and compares their distance in an abstract sense. You can calculate the loss value during parameter update or after updating the parameters too. 

In that cases, you might have to write your own custom loops.

This is my understanding. 

Best Regards.
